src/OrbitGl/CaptureViewElement.cpp

```diff
@@ -60,6 +60,26 @@ CaptureViewElement::EventResult CaptureViewElement::OnMouseWheel(
   return EventResult::kIgnored;
 }
 
+CaptureViewElement::EventResult CaptureViewElement::OnMouseMove(const Vec2& mouse_pos) {
+  if (ContainsPointRecursively(mouse_pos) && !is_mouse_over_) {
+    std::ignore = OnMouseEnter();
+  }
+
+  mouse_pos_cur_ = mouse_pos;
+  // Ignoring the event, so mouse position is updated for its ancestors as well.
```

Comment:
Your point is interesting. I didn't think at this as a possible problem.

- First, you are right. This is out of date as soon the mouse is not over the element anymore (as it's ONLY updated for elements behind the mouse). This can be "fixed" by updating the position to an InvalidMouse position during OnMouseLeave.

- I'm not worried about updating the entire chain (there couldn't be more than 10 updates), but it's true that we might find a solution without using mouse_position.

- It's currently being used in:
--- GlSlider.
--- TrackHeader (to make moving Track works).
--- CallstackThreadBar (to select samples + draw selection).

We can replace it by passing a mouse_position to Draw() methods instead of only passing the timestamp of the mouse. In that way, we can just use mouse_position while drawing (as it's used in the GreenLine and GraphTracks for example).

I suggest to:
- Update mouse_current_pos to an invalid position when MouseLeave.
- Set a TODO to pass it while drawing instead of saving mouse position in each element.
What do you think?

Reply:
SGTM. Thanks a lot for thinking about it.

Reply:
http://b/232530544